app/src/main/java/fr/free/nrw/commons/contributions/ContributionController.java

```diff
@@ -99,6 +215,9 @@ private void setPickerConfiguration(Activity activity,
      */
     private void initiateCameraUpload(Activity activity) {
         setPickerConfiguration(activity, false);
+        if (defaultKvStore.getBoolean("inAppCameraLocationPref", false)) {
+            locationBeforeImageCapture = locationManager.getLastLocation();
```

Comment:
We're presuming the device location would be turned on whenever the setting is turned on. This is incorrect since the user has the freedom to turn on location of their device even after the setting is turned on. 

Ditto on the location permission.

It is prudent to verify that location permission is available and location is turned on whenever the setting is `true`.

Reply:
> We're presuming the device location would be turned on whenever the setting is turned on.

We've already prompted the user to turn it on, redirected to the Settings too.

Reply:
The world is never ideal when it comes to Android. The user can have the setting enabled and then later on turn off location on their device. This is a normal case which we should be wary of. That's precisely why the Nearby feature asks user to turn on location when it is turned off.

The same applies for the location permission (especially on Android 11 and above where permissions are auto-removed for unused apps). So, it is the ideal case to ensure we have location permission and if its unavailable request for the same.

Reply:
A user who wants pictures to be geo-tagged will be turning location on for the camera too but I see it was a bad assumption. Thank you for explaining it! :)